libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java

```diff
@@ -368,6 +371,55 @@ public static boolean contentContainsGutenbergBlocks(String postContent) {
         return (postContent != null && postContent.contains(GUTENBERG_BLOCK_START));
     }
 
+    public static String replaceMediaFileWithUrl(Context context, @NonNull String postContent,
+                                                 String localMediaId, MediaFile mediaFile) {
+        if (mediaFile != null) {
+            String remoteUrl = StringUtils.notNullStr(Utils.escapeQuotes(mediaFile.getFileURL()));
+            // TODO: replace the URL
+            if (!mediaFile.isVideo()) {
+                // replace gutenberg block id holder with serverMediaId, and url_holder with remoteUrl
+                String oldImgBlockHeader = String.format("<!-- wp:image {\"id\":%s} -->", localMediaId);
```

Comment:
Probably we can put this `<!-- wp:image {\"id\":%s} -->` in some constant ?

Reply:
addressed in a321f5e5f1